File: sorting/quicksort.py
def partition(arr,s,e):
    pivot=arr[e]
    i=s-1
    j=s
    while j<=e-1:
        if arr[j]<=pivot:
            i=i+1
            temp=arr[j]
            arr[j]=arr[i]
            arr[i]=temp
        j=j+1
    temp=arr[e]
    arr[e]=arr[i+1]
    arr[i+1]=temp
    return i+1
    

def quicksort(arr,i,j):
    if i>=j:
        return
    p=partition(arr,i,j)
    quicksort(arr,i,p-1)
    quicksort(arr,p+1,j)

File: sorting/QuickSort.java
//Merge Sort
//partion
//divide with pivot
//sort
import java.*;
import java.util.Scanner;  
class QuickSort{
public static void swap(int arr[],int i,int j){
    
    int temp=arr[i];
    arr[i]=arr[j];
    arr[j]=temp;

}
public static int partition(int arr[],int s,int e){
   int i=s-1,j=s;
   while(j<e){
     if(arr[j]<arr[e]){
        i++;
        swap(arr,i,j);
     }
     j++;
   }
   swap(arr,i+1,e);
   return i+1;
}
public static void QuickSort(int arr[],int s,int e){
    if(s>=e){
        return;
    }
    int pivot=partition(arr,s,e);
    QuickSort(arr,s,pivot-1);
    QuickSort(arr,pivot+1,e);
    }

public static int[] getInput(){
    Scanner sc = new Scanner(System.in);
    int n = sc.nextInt();
    int[] arr= new int[n];
    
    for(int i=0;i<n;i++){
        int val = sc.nextInt();
        arr[i]=val;
    }
    return arr;

}

public static void main(String args[]){
    int minidx=0;
    int[] arr=getInput();
    // System.out.print(arr.length);
    QuickSort(arr,0,arr.length-1);
    for(int val:arr){
        System.out.print(val+" ");
    }
   
 }
 
}
